reattacher/reattacher_main.c

```diff
@@ -0,0 +1,172 @@
+/*
+ * Copyright (C) 2020 BiiLabs Co., Ltd. and Contributors
+ * All Rights Reserved.
+ * This is free software; you can redistribute it and/or modify it under the
+ * terms of the MIT license. A copy of the license can be found in the file
+ * "LICENSE" at the root of this distribution.
+ */
+#include <getopt.h>
+#include "accelerator/config.h"
+#include "cclient/api/core/core_api.h"
+#include "cclient/api/extended/extended_api.h"
+#include "common/model/bundle.h"
+#include "storage/ta_storage.h"
+#include "time.h"
+
+#define PRESISTENT_PENDING_SECOND 1800 /**< 30 mins */
+#define DELAY_INTERVAL 300             /**< 5 mins */
+
+#define logger_id scylladb_logger_id
+
+static status_t init_iota_client_service(iota_client_service_t* const serv) {
+  if (serv == NULL) {
+    ta_log_error("Invalid NULL pointer\n");
+    return SC_TA_NULL;
+  }
+  serv->http.path = "/";
+  serv->http.content_type = "application/json";
+  serv->http.accept = "application/json";
+  serv->http.api_version = 1;
+  serv->http.ca_pem = NULL;
+  serv->serializer_type = SR_JSON;
+  if (iota_client_core_init(serv) != RC_OK) {
+    ta_log_error("Failed to connect to IRI.\n");
+    return SC_TA_OOM;
+  }
+  return SC_OK;
+}
+
+static status_t handle_pending_txn(iota_client_service_t* iota_service, db_client_service_t* db_service,
+                                   db_identity_t* obj) {
+  status_t ret = SC_OK;
+  hash243_queue_t req_txn = NULL;
+  get_inclusion_states_res_t* res = get_inclusion_states_res_new();
+  flex_trit_t trit_hash[FLEX_TRIT_SIZE_243];
+  char tryte_hash[NUM_TRYTES_HASH];
+  flex_trits_from_trytes(trit_hash, NUM_TRITS_HASH, (const tryte_t*)db_ret_identity_hash(obj), NUM_TRYTES_HASH,
+                         NUM_TRYTES_HASH);
+
+  hash243_queue_push(&req_txn, trit_hash);
+
+  if (iota_client_get_latest_inclusion(iota_service, req_txn, res) != RC_OK ||
+      get_inclusion_states_res_states_count(res) != 1) {
```

Comment:
May I ask the reason here?
Why could not the state list equals to 1?

Reply:
It should equal to 1 since we only push one transaction into `req_txn`